gems/aws-sdk-cognitoidentity/lib/aws-sdk-cognitoidentity/plugins/wrap_cognito_credentials.rb

```diff
@@ -0,0 +1,52 @@
+module Aws
+  module CognitoIdentity
+    module Plugins
+
+      # This plugin auto populates the following request params for the
+      # CopySnapshot API:
+      #
+      # * `:destination_region`
+      # * `:presigned_url`
+      #
+      # These params are required by EC2 when copying an encrypted snapshot.
+      # @api private
+      class WrapCognitoCredentials < Seahorse::Client::Plugin
+
+        option(
+          :wrap_credentials,
+          default: true,
+          doc_type: 'Boolean',
+          docstring: <<-DOCS)
+When true, wraps the response from get_credentials_for_identity in 
+CognitoIdentityCredentials, which is an auto-refreshing credentials provider
+appropriate for us in other clients.
+        DOCS
+
+        # @api private
+        class Handler < Seahorse::Client::Handler
+
+          def call(context)
+            response = @handler.call(context)
+            params = context.params
+            # TODO: This creates an infite loop
```

Comment:
Let's leave the plugin out of the PR for now..

Reply:
Yeah - agreed for now.  Removing it.